### PR TITLE
adds past date validation

### PR DESCRIPTION
**Pull request by fanatic75**

### CODE REVIEW

File: src/Database/Validator/Datetime.php [hunk 1]
```diff
@@ -11,8 +11,16 @@ class Datetime extends Validator
      */
     protected string $message = 'Date is not valid';
 
-    public function __construct()
+    /**
+     * @var bool
+     */
+    protected bool $allowOldDates = true;
```
Comment (abnegate) [inline src/Database/Validator/Datetime.php:17]: Let's invert this to `$requireDateInFuture = false`, the term `old` doesn't make much sense here without context

File: src/Database/Validator/Datetime.php [hunk 2]
```diff
@@ -37,7 +45,13 @@ public function isValid($value): bool
         }
 
         try {
-            new \DateTime($value);
+            $date = new \DateTime($value);
+            $now = new \DateTime();
+
+            if ($this->allowOldDates === false && $date < $now) {
+                $this->message = 'Date is in the past';
```
Comment (abnegate) [inline src/Database/Validator/Datetime.php:52]: Let's update this too:

```suggestion
                $this->message = 'Date must be in the future';
```

File: src/Database/Validator/Datetime.php (hunk 1 above)
Comment (fanatic75) [reply src/Database/Validator/Datetime.php]: done this

File: src/Database/Validator/Datetime.php (hunk 2 above)
Comment (fanatic75) [reply src/Database/Validator/Datetime.php]: done this

**Review by abnegate — APPROVED** [review]

**Pull request merged**